ASPNET-CORE-MVC/Infrastructure/Services/TransactionService/TransactionService.cs

```diff
@@ -2,87 +2,101 @@
 using Domain.Models;
 using Domain.Repositories;
 using Domain.Services.TransactionService;
-using Infrastructure.Database;
-using Infrastructure.Repositories;
-using ILogger = Serilog.ILogger;
+using Vereyon.Web;
+using System.ComponentModel.DataAnnotations;
 
 namespace Infrastructure.Services.TransactionService;
 
+using ILogger = Serilog.ILogger;
+
 public class TransactionService : ITransactionService
 {
-    private readonly ITransactionRepository _transactionRepository;
+    private readonly IUnitOfWork _unitOfWork;
     private readonly ILogger _logger;
-    private readonly AppIdentityDbContext _context;
+    private readonly IFlashMessage _flashMessage;
 
-    public TransactionService(AppIdentityDbContext context, ILogger logger)
+    public TransactionService(IUnitOfWork unitOfWork, ILogger logger, IFlashMessage flashMessage)
     {
-        _context = context;
-        _logger = logger;
-        _transactionRepository = new TransactionRepository(_context);
+        this._unitOfWork = unitOfWork;
+        this._logger = logger;
+        this._flashMessage = flashMessage;
     }
 
     public async Task<ICollection<TransactionDto>> GetUserTransactions(Guid UserId)
     {
-        try
+        var transactions = await this._unitOfWork.Transactions.Find(c => c.UserId == UserId);
+        if (transactions == null)
```

Comment:
If I am not mistaken, Find always returns Collection and will just return empty collection with count=0, and this block won`t be executed

Reply:
Yes, You are not mistaken. My bad. Fixed. You should consider fixing that mistake in your code too
![image](https://github.com/HoshkoS/F12_2/assets/61410143/dec49b46-b77a-4ae0-87b2-5dc2e34c6f18)